streams/shared/src/main/scala/zio/stream/internal/ChannelExecutor.scala

```diff
@@ -535,15 +536,20 @@ private[zio] class ChannelExecutor[Env, InErr, InElem, InDone, OutErr, OutElem,
     ChannelState.Read(
       self.upstreamExecutor,
       onEffect = (effect: ZIO[Env, Nothing, Unit]) => {
-        val closeLast =
-          if (closeLastSubstream eq null) ZIO.unit else closeLastSubstream
-        closeLastSubstream = null
+        val closeLast = ZIO.suspendSucceed {
+          val close = this.closeLastSubstream
+          closeLastSubstream = null
+          if (close ne null) close else ZIO.unit
```

Comment:
let's be consistent and use `Exit.unit` if that's the style we want to encourage

Reply:
Ah missed this one, thanks!